src/main/webapp/go.js

```diff
@@ -0,0 +1,141 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const GO_URL = '/go-data';
+
+const INVISIBLE_CLASS = 'slide-invisible';
+const NAME_URL = '/name-data';
+const START_ID = 'start-button';
+const PROCEED_ID = 'proceed-button';
+const RIDDLE_ID = 'riddle-area';
+const INDEX_PARAM = 'new-index';
+const FINAL_MSSG = 'Congrats, you\'ve finished the hunt!';
+
+const riddleArr = [];
+let destIndex;
+const destArr = [];
+
+window.onload = getHunt();
+
+/**
+ * Retrieves scavenger hunt data, and updates to the current destination
+ * to reflect the current state of the hunt.
+ */
+function getHunt() {
+  fetch(GO_URL).then((response) => response.json()).then((mssg) => {
+    destIndex = mssg.index;
+    for (let i = 0; i < mssg.items.length; i++) {
+      riddleArr.push(mssg.items[i].riddle.puzzle);
+      destArr.push(mssg.items[i].name);
+    }
+    updateToCurrentState(destIndex);
+  });
+}
+
+/**
+ * Updates the hunt to the current destination that the user is on.
+ *
+ * @param {int} index The index of the destination that the user needs
+ * to find.
+ */
+function updateToCurrentState(index) {
+  if (index >= 0) {
+    hideStartButton();
+    changeRiddleMessage(riddleArr[index]);
+    getDestName();
+  }
+  sendIndexToServlet(index);
+  toggleProceedButton(/* hide = */ true);
```

Comment:
but if index >= 0, after it checks the destination name, if it matches it will show the proceed button.  but then you always hide the proceed button after this if statement.

should this second part only be fore index < 0?

since getDestName is using a fetch, you might need to wait for the promise to resolve.  what happens if you make getDestName async and await it?